src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -170,9 +175,56 @@ export class MicrobitSimulator extends React.Component<any, IState> {
         event.preventDefault();
         this.handleButtonClick(key, true);
     };
-
     protected onMouseLeave = (event: Event, key: string) => {
         event.preventDefault();
         console.log(`To implement onMouseLeave ${key}`);
     };
+    protected onKeyEvent(event: KeyboardEvent, active: boolean, key: string) {
+        event.stopPropagation();
+        if ([event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.ENTER)) {
+            this.handleButtonClick(key, active);
+            if (this.imageRef.current) {
+                if (key === BUTTONS_KEYS.BTN_A) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_A,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_B) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_B,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_AB) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_AB,
+                        active
+                    );
+                }
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.A)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_A, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_A,
+                    active
+                );
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.B)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_B, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_B,
+                    active
+                );
+            }
+        } else if (event.key === CONSTANTS.KEYBOARD_KEYS.CAPITAL_F) {
+            this.togglePlayClick();

```

Comment:
Toggling doesn't seem to work for me. It looks like it toggles twice (E.g. initially it's on: on -> off -> on, instead of on -> off).
Am I missing something?

Reply:
Fixed !

Reply:
This still isn't working for me reliably. Try clicking A/B and then try toggling.

Reply:
Fixed by Andrea! :D